File: JavaScript/7js 1/countChar.js
/*
|--------------------------------------------------------------------------
| Count the Characters
|--------------------------------------------------------------------------
|
*/

function countChar(string, char) {
   return string.toLowerCase().split('').filter(i => i == char.toLowerCase()).length
}


function countChar(string, char) {
   return (string.match(new RegExp(char, 'ig')) || []).length
}


function countChar(s, c) {
   return s.split(new RegExp(c, "i")).length - 1
}


function countChar(s, c) {
   return eval(`(s.match(/${c}/gi)||[]).length`);
}


console.log(countChar("fizzbuzz", "z"), 4);

File: Python/7py 1/count_char.py
"""
|--------------------------------------------------------------------------
| Count the Characters
|--------------------------------------------------------------------------
|
"""

def count_char(s, c):
   return len(list(filter(lambda i: i == c.lower(), s.lower())))


def count_char(s,c):
   return s.lower().count(c.lower())


def count_char(s,x):
   return sum(1 for i in s if i.lower()==x.lower())


print(count_char("Hello there", "e"), 3)
print(count_char("Hello there", "t"), 1)
print(count_char("Hello there", "h"), 2)
print(count_char("Hello there", "L"), 2)
print(count_char("Hello there", " "), 1)